physical/raft/raft.go

```diff
@@ -925,12 +943,31 @@ func (b *RaftBackend) Get(ctx context.Context, path string) (*physical.Entry, er
 	b.permitPool.Acquire()
 	defer b.permitPool.Release()
 
-	return b.fsm.Get(ctx, path)
+	entry, err := b.fsm.Get(ctx, path)
+	if entry != nil {
+		valueLen := len(entry.Value)
+		if uint64(valueLen) > b.kvMaxValueSize {
+			b.logger.Warn(
+				"retrieved entry value is too large; see https://www.vaultproject.io/docs/configuration/storage/raft#raft-parameters",
+				"size", valueLen, "suggested", b.kvMaxValueSize,
+			)
+		}
+	}
+
+	return entry, err
 }
 
-// Put inserts an entry in the log for the put operation
+// Put inserts an entry in the log for the put operation. It will return an
+// error if the value exceeds the configured kv_max_value_size or if the call to
+// applyLog fails.
 func (b *RaftBackend) Put(ctx context.Context, entry *physical.Entry) error {
 	defer metrics.MeasureSince([]string{"raft-storage", "put"}, time.Now())
+
+	valueLen := len(entry.Value)
+	if uint64(valueLen) > b.kvMaxValueSize {
+		return fmt.Errorf("%s; got %d bytes, max: %d bytes", physical.ErrValueTooLarge, valueLen, b.kvMaxValueSize)
```

Comment:
In here and elsewhere, you might want to wrap the error with `errwrap.Wrapf` so that we can type-check with the `errwrap` package (e.g. with `ContainsType`) if need be.

Reply:
Of course, but there is no error here, so what do we wrap? Correct me if I'm wrong :)

Reply:
I see. I misunderstood, thinking that `pnysical.ErrValueTooLarge` was an error type, but it's a string const. This should be good then!